network/servicenetwork/serialize.go

```diff
@@ -52,39 +52,88 @@ package servicenetwork
 
 import (
 	"bytes"
-	"encoding/gob"
 	"io"
 	"io/ioutil"
 
 	"github.com/ThreeDotsLabs/watermill/message"
 	"github.com/pkg/errors"
+
+	"github.com/insolar/insolar/insolar/bits"
 )
 
-// serializeMessage returns io.Reader on buffer with encoded message.Message (from watermill).
-func serializeMessage(msg *message.Message) (io.Reader, error) {
+func serializeString(buffer *bytes.Buffer, s string) {
```

Comment:
i wonder why you didn't check msgpack serialization instead of homemade, maybe add benchmarks?

Reply:
msgpack does not guarantee map keys order in serialized message: https://github.com/msgpack/msgpack/issues/215 
The question is: are there any plans to include message in hash/signature calculation? If no, then there is no need for custom serialization, msgpack is x2 faster. If yes, then there is an option to make keys sorted in custom serialization by adding 3 additional lines of code - and the serialization will be stable (in contrast to msgpack)

Reply:
from my point of view, there is no point of stable serialization for metadata. So it's better just to use protobuf, thanks for idea https://github.com/insolar/insolar/pull/3058

Reply:
replaced with protobuf